pkg/genlib/generator_interface.go

```diff
@@ -839,14 +845,20 @@ func bindNearTimeWithReturn(fieldCfg ConfigField, field Field, fieldMap map[stri
 	var emitF EmitF
 	emitF = func(state *genState) any {
 		var offset time.Duration
-		if fieldCfg.Period > 0 {
+		if fieldCfg.Period > 0 && state.totEvents > 0 {
```

Comment:
can we avoid the repetition here? (it looks identical to the section above, is it?)

Reply:
do you mean in the other function `bindNearTime` at line 498?
every "bind*" function is repeated, once to write to a buffer, once ("bind*WithReturn") to return a value, according to the template type we use (placeholder vs text go template).

If you think it is worth I can try to extract the common logic where possible and just encapsulate a subcall in the two different version of the "bind*" functions, but I suspect it will have impact on performance, that are a sensible topic for the placeholder template (where we look for the highest throughput possible trading off on template features)

Reply:
yeh, the whole section from 501-515.

the logic here is kinda picky, i would much rather not see it repeated. what are the perf implications? isn't it just the overhead of a function call?

Reply:
> what are the perf implications? isn't it just the overhead of a function call?

yes, but that could me too much enough

the placeholder template is more or less micro-optimised for performance and high throughput

see https://github.com/elastic/elastic-integration-corpus-generator-tool/blob/main/docs/performances.md for context

Reply:
> yes, but that could be too much enough

run a benchmark before and after, and in the end it isn't, so I've extracted the logic :)

Reply:
awesome, thanks! great that we can easily benchmark this code to test things like this